src/TypeParser/PhpTypeParser.php

```diff
@@ -56,14 +57,25 @@ public function parseAnnotationType(string $rawType, \ReflectionClass $declaring
             }
         }
 
-        if (0 === \count($types)) {
+        $isCollection = false;
+        $filteredTypes = [];
+        foreach ($types as $type) {
+            $resolvedClass = $this->resolveClass($type, $declaringClass);
+            if (Collection::class === $resolvedClass || is_subclass_of($resolvedClass, Collection::class)) {
```

Comment:
we could use `is_a` here, with 3rd parameter `true` to avoid needing 2 conditions: https://3v4l.org/1v6N1

Reply:
I somehow missed that there is a third parameter. Fixed via 0af690fe5d8e870c1fb7158414f78df3ad8e46ba